ngraph/core/src/op/binary_convolution.cpp

```diff
@@ -73,80 +73,118 @@ op::v1::BinaryConvolution::BinaryConvolution(const Output<Node>& data,
-    const PartialShape& filters_shape = get_input_partial_shape(1);
+    const PartialShape& filters_pshape = get_input_partial_shape(1);
 
-    PartialShape result_shape = PartialShape::dynamic();
-    if (data_batch_shape.rank().is_static())
-    {
-        result_shape =
-            std::vector<Dimension>(data_batch_shape.rank().get_length(), Dimension::dynamic());
+    NODE_VALIDATION_CHECK(this,
+                          data_batch_et.is_real(),
+                          "Data batch element type must be float point. Got: ",
+                          data_batch_et);
 
-        if (data_batch_shape.rank().get_length() > 1)
-        {
-            result_shape[0] = data_batch_shape[0]; // batch size
-        }
+    // TODO: Add NodeValidationCheck to filters et once u1 is supported in nGraph Python API
+    // (#49517)
 
-        if (filters_shape.rank().is_static() && filters_shape.rank().get_length() > 1)
-        {
-            result_shape[1] = filters_shape[0]; // filter channel size
-        }
-    }
+    NODE_VALIDATION_CHECK(this,
+                          data_batch_pshape.rank().compatible(filters_pshape.rank()),
+                          "Shapes for data batch and filters must have same rank. Got: ",
+                          data_batch_pshape,
+                          "and ",
+                          filters_pshape);
 
     if (m_strides.size() == 0)
     {
-        m_strides = conv_default_strides(this, data_batch_shape, filters_shape);
+        m_strides = conv_default_strides(this, data_batch_pshape, filters_pshape);
     }
 
     if (m_dilations.size() == 0)
     {
-        m_dilations = conv_default_strides(this, data_batch_shape, filters_shape);
+        m_dilations = conv_default_strides(this, data_batch_pshape, filters_pshape);
     }
 
     if (m_pads_begin.size() == 0)
     {
-        m_pads_begin = conv_default_padding(this, data_batch_shape, filters_shape);
+        m_pads_begin = conv_default_padding(this, data_batch_pshape, filters_pshape);
     }
 
     if (m_pads_end.size() == 0)
     {
-        m_pads_end = conv_default_padding(this, data_batch_shape, filters_shape);
+        m_pads_end = conv_default_padding(this, data_batch_pshape, filters_pshape);
     }
 
-    if (m_auto_pad == PadType::SAME_UPPER || m_auto_pad == PadType::SAME_LOWER)
+    PartialShape result_shape = PartialShape::dynamic();
+    if (data_batch_pshape.rank().is_static() || filters_pshape.rank().is_static())
     {
-        bool auto_padding_applied = false;
-        if (filters_shape.is_static())
+        const bool is_data_batch_ps_static = data_batch_pshape.rank().is_static();
+        const auto output_ps_rank =
+            is_data_batch_ps_static ? data_batch_pshape.rank() : filters_pshape.rank();
+        const auto num_spatial_dims = is_data_batch_ps_static
+                                          ? data_batch_pshape.rank().get_length() - 2
+                                          : filters_pshape.rank().get_length() - 2;
```

Comment:
Change to `const auto num_spatial_dims = output_ps_rank.get_length() - 2;`

Reply:
Done